src/aclk/https_client.c

```diff
@@ -214,6 +222,10 @@ static int process_http_hdr(http_parse_ctx *parse_ctx, const char *key, const ch
         }
         return 0;
     }
+    void *prev_val = NULL;
+    if (!c_rhash_get_ptr_by_str(parse_ctx->headers, key, &prev_val))
+        freez(prev_val); // drop previous allocation before overwriting
```

Comment:
@stelfrag if you don't want to keep the rest, the above 2 lines are the actual leak.

All the others are supposed to make this more robust.